### ISSUE TITLE
Report/Render cannot be found when publish

### ISSUE THREAD

**Issue by tonysader** [volkanceylan/Serenity#4365]
When I debug my project.. the pdf report works very good but when I publish the project and run it from (iis)  as website
it shows me error 404 
No webpage was found for the web address: http://***.***.*.***:****/Report/Render

**Comment by volkanceylan** [comment]
web.config, appSettings:
```
<add key="EnvironmentSettings" value="{ SiteExternalUrl: 'http://yoursiteurl.com/' }" />
```

**Comment by tonysader** [comment]
Web.config:

`<?xml version="1.0" encoding="utf-8"?>
<configuration>
  
  <system.webServer>
    <handlers>
      <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
    </handlers>
    <aspNetCore processPath="dotnet" arguments=".\Maronite.Web.dll" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" forwardWindowsAuthToken="false" startupTimeLimit="3600" requestTimeout="23:00:00" />
  </system.webServer>
**<appSettings>
<add key="EnvironmentSettings" value="{ SiteExternalUrl: 'http://192.168.1.250:8080/'}"/>
</appSettings>**
</configuration>`

appSettings.json:

`{
  "Data": {
    "Default": {
      "ConnectionString": "******************",
      "ProviderName": "System.Data.SqlClient"
    },
    "***": {
      "ConnectionString": "****************************",
      "ProviderName": "System.Data.SqlClient"
    }
  },
  "Logging": {
    "IncludeScopes": false,
    "LogLevel": {
      "Default": "Debug",
      "System": "Information",
      "Microsoft": "Information"
    }
  },
  "AppSettings": {
    "LocalTextPackages": {
      "Site": [ "Controls.", "Db.", "Dialogs.", "Enums.", "Forms.", "Permission.", "Site.", "Validation." ],
      "Login": [ "Forms.Membership.Login.", "Db.Administration.User.", "Validation.Required", "Dialogs." ]
    },
    "CssBundling": {
      "Enabled": false,
      "Minimize": true,
      "UseMinCss": true
    },
    "ScriptBundling": {
      "Enabled": false,
      "Minimize": true,
      "UseMinJS": true
    },
    "UploadSettings": {
      "Path": "~/App_Data/upload/"
    },
    **"EnvironmentSettings": {
      "SiteExternalUrl": "192.168.1.250:8080/"
    }**
  }
}`

and still the same problem:
This 192.168.1.250 page can't be found
Mp wen page for the web address: http//192.168.1.250:8080/Report/Render

Although before I publish my project when I debug it,  the externalUrl var in the report controller:
` var externalUrl = Config.Get<EnvironmentSettings>().SiteExternalUrl ??
                Request.GetBaseUri().ToString();`

is getting the right value but when I publish it can't find the report controller
@volkanceylan

**Comment by tonysader** [comment]
but when I try to open 
192.168.1.250:8080/Report/Render?key=MyReportKey

it opens the page but without the data

**Comment by tonysader** [comment]
and in appSettings.json  inside the AppSettings section  I added:

```
,
    "EnvironmentSettings": {
      "SiteExternalUrl": "192.168.1.250:8080/"
    }

```

and still have the same problem:
This 192.168.1.250 page can't be found
No webpage was found for the web address: http://192.168.1.250:8080/Report/Render


**Although** before I publish my project when I debug it,  the externalUrl var in the report controller:
```
var externalUrl = Config.Get<EnvironmentSettings>().SiteExternalUrl ??
                Request.GetBaseUri().ToString();
```

is getting the right value but when I publish the project can't find the page 


PS: when I try to open:

192.168.1.250:8080/Report/Render?key=MyReportKey

in the browser it opens the page but without the data

@volkanceylan

**Comment by FinlanStudios** [comment]
@tonysader You ever find out what was causing this issue? I'm running into this problem right now as well.

**Comment by volkanceylan** [comment]
192.168.1.250 is a local address not an internet address.

**Comment by tonysader** [comment]
It’s the address of my server which I am running the project from

**Comment by tonysader** [comment]
Local server yes but what should I do ? @volkanceylan

**Comment by bigjezza** [comment]
I was seeing this publishing to Azure. I resolved it by setting my reports' CSHTML file properties in Visual Studio to 
Build Action ->
![123](https://user-images.githubusercontent.com/71537925/95572966-4a0f5380-0a72-11eb-8b28-3cbd41c34ed8.JPG)
 Content

**Comment by timsan7sk** [comment]
@tonysader @bigjezza a have same issue, posted a request for help on [discussions ](https://github.com/serenity-is/Serenity/discussions/6302#discussion-3912790)
@volkanceylan thx man, you doing a incredible job, I recently met your framework and I'm impressed by its functionality and flexibility.  I already broke my head, did everything that is said in this topic, and although I don't have Azure, but IIS8.5, I took the advice of @bigjezzan, many thanks to him for it.
Has anyone managed to solve this issue?
